### PR TITLE
EDSC-3971: Adding Feature Toggle for cmr-ordering and the EDSC database during down-time

### PR DESCRIPTION
**Pull request by eudoroolivares2016**
# Overview

### What is the feature?

Add feature toggle for cmr-ordering so that we can safely migrate that database
Adds feature toggle for behavior that requires the EDSC database

### What is the Solution?

Adding the feature toggles and disabling behavior from EDSC if they are set to true

### What areas of the application does this impact?

ordering for the order feature toggle and all database interactions as this disabled the secondary toolbar

# Testing

Set the toggles in the process.env in your `.zshrc` or your `.bashrc` for the database toggle and set the toggles for cmr-ordering and the database related rendered components in your `override.static.config.json` file

Ensure your database is either deleted or otherwise turned off I did so using: `pg_ctl -D /usr/local/var/postgresql@14 stop`

To test colormaps toggle against prod locally use: `C1996881146-POCLOUD`

To test cmr-ordering toggle against prod locally use: `C1299783631-LPDAAC_ECS`

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by dpesall — APPROVED** [review]

File: bin/deploy-bamboo.sh [hunk 1]
```diff
@@ -79,8 +81,6 @@ dockerRun() {
         -e "CLOUDFRONT_BUCKET_NAME=$bamboo_CLOUDFRONT_BUCKET_NAME" \
         -e "COLORMAP_JOB_ENABLED=$bamboo_COLORMAP_JOB_ENABLED" \
         -e "DB_INSTANCE_CLASS=$bamboo_DB_INSTANCE_CLASS" \
-        -e "GEOCODING_INCLUDE_POLYGONS=$bamboo_GEOCODING_INCLUDE_POLYGONS" \
```
Comment (eudoroolivares2016) [inline bin/deploy-bamboo.sh:82]: These are no longer being used and should have been removed in a previous PR

File: serverless/src/getColorMap/handler.js [hunk 2]
```diff
@@ -40,6 +41,20 @@ export default async function getColorMap(event, context) {
       body: JSON.stringify({ errors: [`ColorMap '${providedProduct}' not found.`] })
     }
   } catch (error) {
+    const regex = /connect ECONNREFUSED/
+    const dbConnectionError = regex.test(error.message)
+    if (dbConnectionError && disableDatabase) {
+      const colorMapsDisabledMessage = 'Colormaps are currently disabled during maintenance period'
```
[the remark this answers is not in the document]
Comment (dpesall) [reply serverless/src/getColorMap/handler.js:47]: I think it's pretty clear, but you could probably remove the "currently" so it just says "Color maps are disabled during maintenance period." I think the greyed out functionality and the message makes the "currently" unnecessary.

**Review by dpesall — APPROVED** [review]

File: serverless/src/getColorMap/handler.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (eudoroolivares2016) [reply serverless/src/getColorMap/handler.js:47]: The greyed out functionality? Do you mean for the `shapefile` button? In this case the `colorMap` we are handling it as a custom message on the red banner

[the remark this answers is not in the document]
Comment (dpesall) [reply serverless/src/getColorMap/handler.js:47]: Oh I had that mixed up. Either way I think currently is probably a redundant term

File: serverless/src/getColorMap/handler.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { parseError } from '../../../sharedUtils/parseError'
  */
 export default async function getColorMap(event, context) {
   const { defaultResponseHeaders } = getApplicationConfig()
+  const { disableDatabase } = process.env
```
[the remark this answers is not in the document]
Comment (eudoroolivares2016) [reply serverless/src/getColorMap/handler.js]: Removed this

**Review by daniel-zamora — APPROVED** [review]

File: serverless/src/getColorMap/handler.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (eudoroolivares2016) [reply serverless/src/getColorMap/handler.js]: I removed this custom message in favor or just using status app

**Pull request merged**